src/TodoContainer.js

```diff
@@ -0,0 +1,264 @@
+import { useState, useEffect } from "react";
+import App from "./App";
+
+const todaysDate = new Date().toISOString().split("T")[0];
+// ;
+// const day = new Date().getDay();
+
+const TodoContainer = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [masterList, setMasterList] = useState([]);
```

Comment:
Having three different lists here seems dangerous. It's probably too late to change to a different system though. Make sure that when you run your fetch requests for PUT, GET, DELETE, PATCH, etc that you update the todo object in both your masterList and whichever of the other two lists you find it in. Since I assume your item exists in both, you need to make sure it's the exact same in both!

Reply:
A possibly more reliable way of doing this (if you have time) would be to keep one master list and just add a property to the object (and a column in Airtable) for what type of todo it is. Or, better yet, since the date is all that matters to determine if it is upcoming, today, or overdue, you could keep everything in one masterList and in App.js (when you pass a list to each route), create a new list that filters out todos that fit in the other lists and pass that to each list component.

i.e.

```
const overdueList = masterList.filter(todo -> new Date(todo.deadline) < new Date());
```

And similar for the list of today's todos and upcoming todos. You will have to research how to use the Date() API a bit for these comparisons, but it would be a good exercise!